### PR TITLE
`release-libs.yaml` automatically triggered on PRs to `main`

### PR DESCRIPTION
**Pull request by plebhash**
Every time we merge some PR from `dev` to `main`, we need to make sure every modified crate version is properly increased. Otherwise, we will mess up the dependency chain of whoever is fetching from crates.io

This PR:
- changes `release-libs.yaml` so that the `Release Libs` workflow is triggered every time a PR is created against `main`.
- introduces a shell script (`check-versioning-lib-release.sh`) that checks the diff between `main` and `dev` branches... if some lib crate has been changed but its version wasn't bumped, the CI will trigger an error and stop (here's an [example](https://github.com/plebhash/stratum/actions/runs/8530384769/job/23368058479?pr=47#step:3:15) on my fork)
- removes `roles` from being published to crates.io (it doesn't make sense until https://github.com/stratum-mining/stratum/issues/702 is completed, which will take a long time)

### CODE REVIEW

**Comment by Fi3** [comment]
It would be nice to have a script that look at diff with main and for the crates that do diff, it run cargo release --dry run. So that we can run the script as an action before merging to main and spot if we forgot to change a version of a changed crate. (less thing to think about when we do release that is not bad)

**Comment by plebhash** [comment]
> It would be nice to have a script that look at diff with main and for the crates that do diff, it run cargo release --dry run. So that we can run the script as an action before merging to main and spot if we forgot to change a version of a changed crate. (less thing to think about when we review release that is not bad)

that's a good idea, I'll write a new script and add to this PR

**Comment by plebhash** [comment]
> It would be nice to have a script that look at diff with main and for the crates that do diff, it run cargo release --dry run. So that we can run the script as an action before merging to main and spot if we forgot to change a version of a changed crate. (less thing to think about when we review release that is not bad)

I wrote a shell script on this commit https://github.com/stratum-mining/stratum/pull/790/commits/83733a86b4b85eab202c8447a8debfc3dbd13481

however, it is purely shell scripting + git, it does not really use `cargo release` in any way

**Review by GitGab19 — APPROVED** [review]
Great idea to have a script to check that before merging `dev` into `main` 👏

File: .github/workflows/release-libs.yaml
Comment (plebhash) [inline .github/workflows/release-libs.yaml:157]: these are being removed because it doesn't make sense to publish roles to `crates.io` before https://github.com/stratum-mining/stratum/issues/702 is solved

and that will probably take a long time

File: check-versioning-lib-release.sh
Comment (plebhash) [inline check-versioning-lib-release.sh:51]: here's an example of how this shell script will prevent us from merging to `main` if we forget to bump some crate's version:

https://github.com/plebhash/stratum/actions/runs/8530384769/job/23368058479?pr=47#step:3:15

**Comment by plebhash** [comment]
`release-libs.yaml` is failing on CI with this error message:
```
please paste the token found on https://crates.io/me below
error: credential provider `cargo:token` failed action `login`

Caused by:
  please provide a non-empty token
Error: Process completed with exit code 101.
```

I suspect this is because this workflow is being triggered from my fork, which doesn't have the login info to crates.io configured.

Also, this failed workflow is not blocking the PR from being merged.

IMO I would ignore this and merge (if reviews ok). After merging, I can do a dummy PR from `dev` to `main`  to see if this error remains (I suspect it won't).

**Review by GitGab19 — APPROVED** [review]

**Pull request merged**
